### ISSUE TITLE
Problems with syntax conceal in preview

### ISSUE THREAD

**Issue by karamellpelle** [luc-tielen/telescope_hoogle#12]
Hi! I have two problems with this plugin regarding syntax concealing:

0. **The Telescope preview window does not conceal html, i.e. it shows html tags**.
   Not sure why [these lines does not work for me](https://github.com/luc-tielen/telescope_hoogle/blob/5c2ae51bcf905a7101134a597e6f7be2dc05f975/lua/telescope/_extensions/hoogle/live_hoogle_search.lua#L40C1-L46C7). However, if I add `require("notify")("hello")` (using the [nvim-notify](https://github.com/rcarriga/nvim-notify) plugin) at the end of these lines, everything works. I found this side effect while trying to debug. Maybe something needs to be flushed/redrawn?
   
   It does neither work if I add `setlocal conceallevel=2` (as many regular syntax definition files does) in [telescope_hoogle_doc.vim](https://github.com/luc-tielen/telescope_hoogle/blob/5c2ae51bcf905a7101134a597e6f7be2dc05f975/syntax/telescope_hoogle_doc.vim). 

   Everything works if I have global conceal enabled (`set conceallevel=2` in command mode)
1. **The rules have problems with some hoogle results**. Search after `Fractional` and there are some `<b>` tags shown. For most results the rules work well :)

Thanks for your plugin!

------

Here is the Hoogle HTML result for Fractional:

~~~html
Fractional numbers, supporting real division.

The Haskell Report defines no laws for <a>Fractional</a>. However,
<tt>(<a>+</a>)</tt> and <tt>(<a>*</a>)</tt> are customarily expected
to define a division ring and have the following properties:

<ul>
<li><i><b><a>recip</a> gives the multiplicative inverse</b></i> <tt>x
* recip x</tt> = <tt>recip x * x</tt> = <tt>fromInteger 1</tt></li>
<li><i><b>Totality of <a>toRational</a></b></i> <a>toRational</a> is
total</li>
<li><i><b>Coherence with <a>toRational</a></b></i> if the type also
implements <a>Real</a>, then <a>fromRational</a> is a left inverse for
<a>toRational</a>, i.e. <tt>fromRational (toRational i) = i</tt></li>
</ul>

Note that it <i>isn't</i> customarily expected that a type instance of
<a>Fractional</a> implement a field. However, all instances in
<tt>base</tt> do.
~~~

**Comment by galyni** [comment]
@karamellpelle  Same here. Thanks for the `set conceallevel=2` tip, I added that to my config file.

**Comment by karamellpelle** [comment]
Were you able to set conceallevel locally for that buffer only? I haven't tested in a while..

**Comment by galyni** [comment]
No, but reading the code for the previewer, I guess this should set it.

https://github.com/luc-tielen/telescope_hoogle/blob/5c2ae51bcf905a7101134a597e6f7be2dc05f975/lua/telescope/_extensions/hoogle/live_hoogle_search.lua#L34-L47

The fact that we had to set it globally indicates it doesn't work. Will try to do something about that.

**Comment by galyni** [comment]
@karamellpelle 
Well, after some print debugging and telescope code and documentation reading, here's what I can say.

1. the display_contents method, which is used by this extension, has been deprecated by telescope. Maybe the way telescope handle retrocompatibility has broken something here?
2. I am not sure what "bufnr" refers to, but maybe the buffer changes ? See https://github.com/nvim-telescope/telescope.nvim/blob/b4da76be54691e854d3e0e02c36b0245f945c2c7/doc/telescope.txt#L3988-L3992
3. If I do `vim.wo.conceallevel = 2` instead of `vim.wo[win].conceallevel = 2` (and we can do that outside of the `nvim_buf_call`), it works and the conceallevel is not changed globally. 

Not sure what is better: making a PR to just set the options using `vim.wo` or trying to use the new api instead of `display_content`.
@luc-tielen I can try to do either if you have an opinion and have not already plan to migrate away from `display_content`, like the TODO indicates.

**Comment by luc-tielen** [comment]
@galyni Feel free to open a PR. It probably is indeed broken because of the deprecated API since this plugin is quite old by now (and I don't use it actively anymore). Keep in mind that I am busy though so I may be slow to respond.

**Comment by karamellpelle** [comment]
> [@galyni](https://github.com/galyni) (and I don't use it actively anymore). 

What do you use for hoogle lookup / API references now, @luc-tielen  ?

**Comment by luc-tielen** [comment]
@karamellpelle Nothing, I haven't been writing any Haskell lately since I've been busy with work and IRL stuff and my haskell gig ended a while ago. 🥲

But while I was still using it at work, the haskell language server was becoming better and better, so I started using hoogle less (or from time to time directly from the browser).
